RInAppMessaging/Classes/CampaignDispatcher.swift

```diff
@@ -15,17 +15,18 @@ internal protocol CampaignDispatcherType {
 
 /// Class for adding ready (validated) campaigns to a queue to be sequentially displayed.
 /// Each next campaign is scheduled after closing according to the Campaign's delay paramater.
-internal class CampaignDispatcher: CampaignDispatcherType {
+internal class CampaignDispatcher: CampaignDispatcherType, TaskSchedulable {
 
     private let router: RouterType
     private let permissionService: DisplayPermissionServiceType
     private let campaignRepository: CampaignRepositoryType
 
     private let dispatchQueue = DispatchQueue(label: "IAM.Campaign", qos: .userInteractive)
     private var queuedCampaigns = [Campaign]()
-    private(set) var isDispatching = false
+    @AtomicGetSet private(set) var isDispatching = false
 
     weak var delegate: CampaignDispatcherDelegate?
+    var scheduledTask: DispatchWorkItem?
```

Comment:
Where does this get set?
I only see a call to cancel()

Reply:
Ah, it's from TaskSchedulable

Reply:
That's right